docs/developer-guide/architecture-and-design.md

```diff
@@ -19,4 +19,10 @@ group: developer-guide
 
 ## BenchFlow DSL
 
-<!--  -->
\ No newline at end of file
+## Serializing/Deserializing YAML
+To serialize/deserialize YAML the DSL library uses [MoultingYAML](https://github.com/jcazevedo/moultingyaml).
+
+### Naming of `*YamlProtocol` files and case classes
+Most of the `*YamlProtocol` files can be shared for both Tests and Experiments, as well as their case class counterparts. In the cases where we need to separate the protocols we different the files by prepending `BenchFlowTest` and `BenchFlowExperiment` for tests and experiments respectively, e.g. `BenchFlowExperimentConfigurationYamlProtocol` and `BenchFlowExperimentConfiguration`.
```

Comment:
Make it also explicit that nothing is prepended when there is not need to separate files, because they are shared exactly the same between Test and Experiment.